registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -171,171 +168,166 @@ NormalDistributionsTransform<PointSource, PointTarget>::computeTransformation (P
-  // Original Point and Transformed Point (for math)
-  Eigen::Vector3d x, x_trans;
-  // Occupied Voxel
-  TargetGridLeafConstPtr cell;
-  // Inverse Covariance of Occupied Voxel
-  Eigen::Matrix3d c_inv;
-
   score_gradient.setZero ();
   hessian.setZero ();
   double score = 0;
 
   // Precompute Angular Derivatives (eq. 6.19 and 6.21)[Magnusson 2009]
-  computeAngleDerivatives (p);
+  computeAngleDerivatives (transform);
 
   // Update gradient and hessian for each point, line 17 in Algorithm 2 [Magnusson 2009]
-  for (std::size_t idx = 0; idx < input_->points.size (); idx++)
+  for (std::size_t idx = 0; idx < input_->size (); idx++)
   {
-    x_trans_pt = trans_cloud[idx];
+  // Transformed Point
+    const auto& x_trans_pt = trans_cloud[idx];
+    Eigen::Vector3d x_trans = x_trans_pt.getVector3fMap().template cast<double>();
 
-    // Find nieghbors (Radius search has been experimentally faster than direct neighbor checking.
+    // Find neighbors (Radius search has been experimentally faster than direct neighbor checking.
     std::vector<TargetGridLeafConstPtr> neighborhood;
     std::vector<float> distances;
     target_cells_.radiusSearch (x_trans_pt, resolution_, neighborhood, distances);
 
     for (typename std::vector<TargetGridLeafConstPtr>::iterator neighborhood_it = neighborhood.begin (); neighborhood_it != neighborhood.end (); ++neighborhood_it)
     {
-      cell = *neighborhood_it;
-      x_pt = (*input_)[idx];
-      x = Eigen::Vector3d (x_pt.x, x_pt.y, x_pt.z);
+      // Occupied Voxel
+      const auto& cell = *neighborhood_it;
 
-      x_trans = Eigen::Vector3d (x_trans_pt.x, x_trans_pt.y, x_trans_pt.z);
+      // Original Point
+      const auto& x_pt = (*input_)[idx];
+      Eigen::Vector3d x = x_pt.getVector3fMap().template cast<double>();
 
       // Denorm point, x_k' in Equations 6.12 and 6.13 [Magnusson 2009]
       x_trans -= cell->getMean ();
+      // Inverse Covariance of Occupied Voxel
       // Uses precomputed covariance for speed.
-      c_inv = cell->getInverseCov ();
+      Eigen::Matrix3d c_inv = cell->getInverseCov ();
 
       // Compute derivative of transform function w.r.t. transform vector, J_E and H_E in Equations 6.18 and 6.20 [Magnusson 2009]
       computePointDerivatives (x);
       // Update score, gradient and hessian, lines 19-21 in Algorithm 2, according to Equations 6.10, 6.12 and 6.13, respectively [Magnusson 2009]
       score += updateDerivatives (score_gradient, hessian, x_trans, c_inv, compute_hessian);
-
     }
   }
-  return (score);
+  return score;
 }
 
 
 template<typename PointSource, typename PointTarget> void
-NormalDistributionsTransform<PointSource, PointTarget>::computeAngleDerivatives (Eigen::Matrix<double, 6, 1> &p, bool compute_hessian)
+NormalDistributionsTransform<PointSource, PointTarget>::computeAngleDerivatives (Eigen::Matrix<double, 6, 1> &transform, bool compute_hessian)
 {
   // Simplified math for near 0 angles
   double cx, cy, cz, sx, sy, sz;
```

Comment:
If this were 2 arrays, this logic would be converted to a loop. If not, then consider a local lamda instead (to preserve readability in latter sections)

Reply:
I made a local lambda to do the simplified cos/sin calculation.